src/components/MyAccount/RequestsTab/UpdateLocation.tsx

```diff
@@ -0,0 +1,196 @@
+// Modal onClose
+import {
+  Box,
+  Icon,
+  useModal,
+  Text,
+  Heading,
+  Button,
+  Select,
+  Link as DSLink,
+  type DefaultModalProps,
+  type ConfirmationModalProps,
+  type BaseModalProps,
+  SkeletonLoader,
+} from "@nypl/design-system-react-components"
+import type { SierraCodeName } from "../../../types/myAccountTypes"
+import styles from "../../../../styles/components/MyAccount.module.scss"
+import { useState } from "react"
+import { BASE_URL } from "../../../config/constants"
+
+interface UpdateLocationPropsType {
+  holdId: string
+  pickupLocation: SierraCodeName
+  pickupLocationOptions: SierraCodeName[]
+  key: number
+  patronId: number
+  updateHoldLocation: (holdId: string, location: SierraCodeName) => void
+}
+
+const UpdateLocation = ({
+  updateHoldLocation,
+  pickupLocationOptions,
+  patronId,
+  holdId,
+  pickupLocation,
+  key,
+}: UpdateLocationPropsType) => {
+  const { Modal, onOpen: openModal, onClose: closeModal } = useModal()
+  const locationsWithSelectedFirst = [
+    pickupLocation,
+    ...pickupLocationOptions.filter(
+      (loc) => loc.code.trim() !== pickupLocation.code.trim()
+    ),
+  ]
+  const confirmLocationChangeModalProps = (selected: SierraCodeName) => ({
+    type: "confirmation",
+    bodyContent: (
+      <Box className={styles.modalBody}>
+        <Select
+          value={selected.code}
+          onChange={(e: React.FormEvent<HTMLInputElement>) => {
```

Comment:
Does this make the whole modal flash on change bc of the Button Group error ?

Reply:
I'm pretty sure it's because the entire modal props has to be updated so the modal is rerendering. It's the same as when you click confirm and it flashes before showing you success or failure.